pyllamacpp/model.py

```diff
@@ -137,19 +155,11 @@ def generate(self,
         :param verbose: if `True`, `llama.cpp` stuff will be printed
         :return: Tokens generator
         """
-        prompt = f' {self.prompt_prefix}{prompt}{self.prompt_suffix}'
-        input_tokens = pp.llama_tokenize(self._ctx, prompt, True)
-        if len(input_tokens) > self._n_ctx - 4:
+        input_tokens = self._prompt_prefix_tokens + pp.llama_tokenize(self._ctx, prompt, True) + self._prompt_suffix_tokens
+        if len(input_tokens) > self._n_ctx - 4 - len(self._prompt_context_tokens):
             raise Exception('Prompt too long!')
-        predicted_tokens = []
-        predicted_token = 0
-
-        # add global context for the first time
-        if self._n_past == 0:
-            for tok in self._prompt_context_tokens:
-                predicted_tokens.append(tok)
-                self._last_n_tokens.pop(0)
-                self._last_n_tokens.append(tok)
+        predicted_tokens = self._prompt_context_tokens.copy()
+        predicted_token = len(predicted_tokens)
```

Comment:
Why you have initialized the `predicted_token` with `len(predicted_tokens)`, is there any reasoning behind this ?